CHANGELOG.md

```diff
@@ -53,6 +53,7 @@
 * [FEATURE] TLS config options added for GRPC clients in Querier (Query-frontend client & Ingester client), Ruler, Store Gateway, as well as HTTP client in Config store client. #2502
 * [FEATURE] The flag `-frontend.max-cache-freshness` is now supported within the limits overrides, to specify per-tenant max cache freshness values. The corresponding YAML config parameter has been changed from `results_cache.max_freshness` to `limits_config.max_cache_freshness`. The legacy YAML config parameter (`results_cache.max_freshness`) will continue to be supported till Cortex release `v1.4.0`. #2609
 * [FEATURE] Experimental gRPC Store: Added support to 3rd parties index and chunk stores using gRPC client/server plugin mechanism. #2220
+* [ENHANCEMENT] Add command line flag to list possible values for -target. Also, throw error if the given target is not public. #2710
```

Comment:
I think "internal" modules should be selectable, on some occasions it is useful. We can perhaps just log the warning instead?

Reply:
> I think "internal" modules should be selectable, on some occasions it is useful. We can perhaps just log the warning instead?

Yea, if there are cases where we want to use internal modules, then we definitely should just log an error instead of throwing error. I will make this change. 

> I'd prefer to have single RegisterModule method with options, to reduce number of exported methods and structs. What do you think?

I would prefer this too, but I was worry about backward compatibility since RegisterModule is an exported method. Is there anyway to have an optional parameter in Go? Or should I not worry about backward compatibility?